src/redisraft.c

```diff
@@ -413,8 +413,7 @@ static bool handleInterceptedCommands(RedisRaftCtx *rr,
  * 3. If the hash slot is associated with a foreign ShardGroup, perform a redirect.
  * 4. If the hash slot is not mapped, produce a CLUSTERDOWN error.
  */
-static RRStatus handleSharding(RedisRaftCtx *rr,
-                               RedisModuleCtx *ctx, RaftRedisCommandArray *cmds)
+RRStatus handleSharding(RedisRaftCtx *rr, RedisModuleCtx *ctx, RaftRedisCommandArray *cmds)
```

Comment:
right now its more complicated than it needs to be, as its done at both append time and apply time, we should perhaps consider getting rid of the code at append time, which would simplify the function, as wouldn't need to validate shard group (sg) here) as would be validated in the `validateRaftRedisCommandArray()` function call